### DISCUSSION TITLE
Using “post” in Docker action

### DISCUSSION THREAD

**Discussion by zmeyc** [community/community#25371] [Actions]
<p>In nodejs action, it’s possible to use ‘post’ for cleanup:<br>
action.yml</p>
<pre><code class="lang-auto">runs:
  using: 'node12'
  main: 'dist/index.js'
  post: 'dist/index.js'
</code></pre>
<p>index.js:</p>
<pre><code class="lang-auto">const isPost = Boolean(process.env['STATE_isPost'])
</code></pre>
<p>If isPost is true, then it’s a cleanup step.</p>
<p>How to request ‘post’ in Docker action, is it supported?</p>
<pre><code class="lang-auto">runs:
  using: 'docker'
  image: 'Dockerfile'
  ?
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/zmeyc">@zmeyc</a>,</p>
<p>’ <a href="https://help.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#post" rel="nofollow noopener"> <strong><code>post</code></strong> </a>’ is only used for javasript action, for docker action, please use ’ <a href="https://help.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#post-entrypoint" rel="nofollow noopener"> <strong><code>post-entrypoint</code></strong> </a>'.</p>
<pre><code class="lang-auto">runs:
  using: 'docker'
  image: 'Dockerfile'
  args:
  - 'bzz'
  entrypoint: 'main.sh'
  post-entrypoint: 'cleanup.sh'
</code></pre>
<p>Please check the official <a href="https://help.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions" rel="nofollow noopener">doc</a> for more details.</p>
<p>Thanks.</p>
